shavit-bash2.sp

```diff
@@ -2385,20 +2395,31 @@ stock void RecordStartStrafe(int client, int button, int turnDir, const char[] c
 	#endif
 	g_bStartStrafe_IsRecorded[client][currFrame] = true;
 	g_iStartStrafe_CurrentFrame[client] = (g_iStartStrafe_CurrentFrame[client] + 1) % MAX_FRAMES;
-
-	if(g_iStartStrafe_Stats[client][StrafeData_Difference][currFrame] == 0 && !IsInLeftRight(client, g_iRealButtons[client]))
+	
+	
+	if(g_iStartStrafe_Stats[client][StrafeData_Difference][currFrame] == g_iStartStrafe_LastTickDifference[client] && !IsInLeftRight(client, g_iRealButtons[client]))
 	{
-		g_iStartStrafe_PerfCount[client]++;
+		g_iStartStrafe_IdenticalCount[client]++;
 	}
-	else
+	else if (g_iStartStrafe_IdenticalCount[client] >= IDENTICAL_STRAFE_MIN)
+	{	
+		AnticheatLog(client, "too many %i strafes in a row (%d)", g_iStartStrafe_LastTickDifference[client], g_iStartStrafe_IdenticalCount[client]);
+		//AutoBanPlayer(client);
+		
+		g_iStartStrafe_LastTickDifference[client] = g_iStartStrafe_Stats[client][StrafeData_Difference][currFrame];
+		g_iStartStrafe_IdenticalCount[client] = 0;
+	}
+	else if(g_iStartStrafe_IdenticalCount[client] >= 10)
 	{
-		g_iStartStrafe_PerfCount[client] = 0;
+		AnticheatLog(client, "too many %i strafes in a row (%d)", g_iStartStrafe_LastTickDifference[client], g_iStartStrafe_IdenticalCount[client]);
+		
+		g_iStartStrafe_LastTickDifference[client] = g_iStartStrafe_Stats[client][StrafeData_Difference][currFrame];
+		g_iStartStrafe_IdenticalCount[client] = 0;
```

Comment:
You probably don't want to reset these variables here or g_iStartStrafe_IdenticalCount will never hit the autoban limit.